cmd/thanos/sidecar.go

```diff
@@ -457,3 +398,39 @@ func (s *promMetadata) Timestamps() (mint int64, maxt int64) {
 
 	return s.mint, s.maxt
 }
+
+type sidecarConfig struct {
+	grpc         grpcConfig
+	http         httpConfig
+	prometheus   prometheusConfig
+	tsdb         tsdbConfig
+	connection   connectionConfig
+	reloader     reloaderConfig
+	shipper      shipperConfig
+	objStore     extflag.PathOrContent
+	limitMinTime thanosmodel.TimeOrDurationValue
+}
+
+func newSidecarConfig(cmd *kingpin.CmdClause) *sidecarConfig {
+	var grpcConf grpcConfig
```

Comment:
We don't need any of this var definition, we can use directly sidecar config (: 

I think what would look better is just RegisterFlag method for sidecar.


Then we can create sidecar struct and do `RegisterFlags`. Then in that method we can run all RegisterFlags of each field for this struct.  Those method has to have pointer receivers to make a difference.

Reply:
It's bit repetitive, but good enough (: